sematic/ui/packages/common/src/hooks/metricsHooks.ts

```diff
@@ -0,0 +1,71 @@
+import { useHttpClient } from "@sematic/common/src/hooks/httpHooks";
+import useAsync from "react-use/lib/useAsync";
+import { BasicMetricsPayload } from "@sematic/common/src/ApiContracts";
+import { useMemo } from "react";
+import { durationSecondsToString } from "@sematic/common/src/utils/datetime";
+
+export function runSuccessRate(
+    countByState: { [k: string]: number },
+): string {
+    const numeratorStates = ["RESOLVED"];
+    const denominatorStates = ["RESOLVED", "FAILED", "NESTED_FAILED"];
+
+    const countForStates = (states: string[]) =>
+        states.reduce(
+            (sum: number, state: string) => (sum += countByState[state] || 0),
+            0
+        );
+
+    const percentRate =
+        (countForStates(numeratorStates) /
+            (countForStates(denominatorStates) || 1)) *
+        100;
+
+    return `${Math.floor(percentRate)}%`;
+}
+
+export function runAvgRunTime(
+    avgRuntimeChildren: { [k: string]: number },
+    functionPath: string
+): string {
+    return durationSecondsToString(avgRuntimeChildren[functionPath] || 0);
+}
+
+interface useBasicMetricsResult {
+    payload: BasicMetricsPayload | undefined;
+    loading: boolean;
+    error: Error | undefined;
+    totalCount: number | undefined;
+    successRate: string;
+    avgRuntime: string;
```

Comment:
If `totalCount` can be undefined, can't `successRate` and `avgRuntime` also be undefined?

Reply:
`successRate` and `avgRuntime` will have a default value of `0%` and `0s`. So they will not be undefined.